### DISCUSSION TITLE
Is config support for `:has` selectors being considered?

### DISCUSSION THREAD

**Discussion by crswll** [tailwindlabs/tailwindcss#12707] [Ideas]
I'd like to be able to do something like this in `tailwind.config.js`:
```
has: {
  'at-least-3-children': ':nth-child(3)'
}
```
and use it like
```html
<div class="has-at-least-3-children:bg-green-500 bg-red-500">...</div>
```

I considered opening up a PR for this but ran into a few questions.

1. Would `theme` be a good place for those settings? It seems odd there to me but might just be due to some `has` examples I came up with.
2. If you wanted to allow variants to be configured where would you want that? `dark` has it's own special thing and `screen` pulls from `theme` (but it makes sense there).

**Comment by wongjn** [comment]
For what its worth, similar variants that can have an arbitrary portion, like [`aria-*`](https://tailwindcss.com/docs/hover-focus-and-other-states#aria-states) and [`data-*`](https://tailwindcss.com/docs/hover-focus-and-other-states#data-attributes) use `theme.aria` and `theme.data` respectively.

**Reply by crswll** [reply]
Ah, forgot about those! I would have questioned putting those in `theme` as well so maybe `has` would be all right in there and I'm just over thinking it.

**Comment by olets** [comment]
Not strictly speaking a direct answer to your question, but you can do this already with either an [arbitrary variant](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) or a [dynamic variant plugin](https://tailwindcss.com/docs/plugins#dynamic-variants)

https://play.tailwindcss.com/H255ISfixA

**Reply by crswll** [reply]
Thanks! Yeah that was just an example. I opened up a [PR](https://github.com/tailwindlabs/tailwindcss/pull/12711) for it so we'll see what happens with that. Did write [my own plugin](https://play.tailwindcss.com/rjioAJSH7G?file=config) for `:has` a while ago but was hoping to not have to use it.

**Reply by olets** [reply]
I'd be interested to hear more about why you're hoping to not use your plugin.

**Reply by crswll** [reply]
Just less code for me personally to think about.
